packages/playground/data-liberation/src/export/WP_Exporter.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+use WordPress\Zip\ZipStreamWriter;
+
+class WP_Exporter {
+	public static function stream_export( $output_stream = false ) {
+		// @TODO: This is a hack. Maybe we should have a way to export without setting headers.
+		$preexisting_response_headers = headers_list();
+
+		require_once ABSPATH . 'wp-admin/includes/export.php';
+		ob_start();
+		export_wp();
+
+		// @TODO: This is a hack to avoid headers set by export_wp(). Maybe we should have a way to export without setting headers.
+		header_remove();
+		foreach ( $preexisting_response_headers as $header ) {
+			header( $header, false );
+		}
+
+		$wxr_content = ob_get_clean();
+
+		// @TODO: Replace upload URLs with relative file URLs.
+
+		$uploads = wp_upload_dir();
+		// @TODO: This is a hack and kind of broken. Replace attachment URLs using proper XML and URL parsing libraries.
+		$wxr_content = str_replace(
+			trailingslashit( $uploads['baseurl'] ),
+			'file://./wp-content/uploads/',
+			$wxr_content
+		);
+
+		header('Content-Type: application/zip');
+
+		// @TODO: Can we get rid of this open-stdout-on-demand workaround?
+		// NOTE: Opening stdout on demand after output buffering the export
+		// because output buffering seemed to interfere with a preexisting stdout stream.
+		// By opening stdout after output buffering, streaming the zip to stdout appears to work.
+		if ( !$output_stream ) {
+			$output_stream = fopen('php://output', 'wb');
+		}
+		$zip_writer = new ZipStreamWriter( $output_stream );
```

Comment:
Let's try packaging the data using the same `WP_Entity` objects as the importer. We could then build multiple exporting backends, e.g. WXR, Markdown, HTML, etc. but even more importantly, we could serialize the exported data, send It over the wire, and import without an intermediate format. That's important for site sync protocol and for things like the Try WordPress extension. Plus we could extend it to more data types, e.g. SQL dumps, Blueprint steps, etc.

Reply:
✅ @adamziel, I don't know exactly what this means yet but will look at the importer work for reference.

Reply:
I spoke with Adam, and what we are talking about is basically making a WP_Entity iterator API that can be used to read WP entities from a site. Then the entity iterator API can be used to implement multiple exporters.

Reply:
The plan for this PR is to just tweak URL replacement to work properly and then leave open as a draft until it can be replaced with a proper exporter based on the entity iterator.